gh-139707: Add docs for optional modules (#140171)
python/cpython: Doc/includes/optional-module.rst

This is an :term:`optional module`.
If it is missing from your copy of CPython,
look for documentation from your distributor (that is,
whoever provided Python to you).
If you are the distributor, see :ref:`optional-module-requirements`.

> .. Notes similar wording appear in these module docs:

Reviewer: ```suggestion
.. Similar notes appear in the docs of the modules:
```

Author: Oops! Thanks :)
